vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -5,34 +5,53 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import edu.asu.diging.vspace.core.data.BranchingPointRepository;
+import edu.asu.diging.vspace.core.data.ChoiceRepository;
 import edu.asu.diging.vspace.core.data.SlideRepository;
 import edu.asu.diging.vspace.core.factory.impl.SlideFactory;
+import edu.asu.diging.vspace.core.model.IBranchingPoint;
+import edu.asu.diging.vspace.core.model.IChoice;
 import edu.asu.diging.vspace.core.model.IModule;
 import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.impl.BranchingPoint;
+import edu.asu.diging.vspace.core.model.impl.Choice;
 import edu.asu.diging.vspace.core.model.impl.Slide;
 import edu.asu.diging.vspace.core.services.ISlideManager;
 import edu.asu.diging.vspace.web.staff.forms.SlideForm;
 
 @Service
 public class SlideManager implements ISlideManager {
 
+    @Autowired
+    private SlideFactory slideFactory;
+ 
     @Autowired
     private SlideRepository slideRepo;
-
+    
     @Autowired
-    private SlideFactory slideFactory;
-
+    private BranchingPointRepository bpointRepo;
+    
     @Autowired
-    private ModuleManager moduleManager;
+    private ChoiceRepository choiceRepo;
+ 
 
     @Override
-    public ISlide createSlide(String moduleId, SlideForm slideForm) {
-        IModule module = moduleManager.getModule(moduleId);
-        ISlide slide = slideFactory.createSlide(module, slideForm);
-        slideRepo.save((Slide) slide);
+    public ISlide createSlide(IModule module, SlideForm slideForm) {
+        ISlide slide = slideFactory.createSlide(module, slideForm);             
+        slideRepo.save((Slide) slide);        
         return slide;
     }
 
+    @Override
+    public IBranchingPoint createBranchingPoint(IModule module, SlideForm slideForm) {
+        IBranchingPoint branchingPoint = slideFactory.createBranchingPoint(module, slideForm); 
+        for(IChoice choice: branchingPoint.getChoices()) {
+            choiceRepo.save((Choice)choice);
+        }           
```

Comment:
I forgot, have you tried if this works without explicitly saving the choices?

Reply:
it worked when used cascade=CascadeType.ALL. Which way should I go with?

Reply:
since the choices without a branching point don't make sense, I'd cascade them all and avoid saving them individually.